### ISSUE TITLE
Raster.reproject not working properly when only giving dst_bounds

### ISSUE THREAD

**Issue by erikmannerfelt** [GlacioHack/GeoUtils#140]
I tried loading and cropping a xdem DEM (subclass of Raster) in one line:

```python
bounds = rio.coords.BoundingBox(493927, 8768560, 501145, 8776349)
  
dem2011 = xdem.dem.DEM(dem_paths[2011]).reproject(dst_bounds=bounds)
```
The 2011 DEM in question is huge and the `bounds` object is a very small subset relatively speaking. Still, this warning is triggered:
```
/home/erik/.local/share/conda/xdem/lib/python3.8/site-packages/geoutils/georaster.py:717: UserWarning: Output projection, bounds and size are identical -> return self (not a copy!)
  warnings.warn("Output projection, bounds and size are identical -> return self (not a copy!)")
```

I am using the latest geoutils and xdem (installed by pip install git+...)

**Comment by erikmannerfelt** [comment]
Specifying a `dst_crs` CRS made this work. I coincidentally also wanted to convert ETRS89 UTM 33N to WGS84 UTM 33N, and adding this fixed the bounds issue as well.

**Comment by atedstone** [comment]
Cropping was not originally considered as a use case for `reproject` - it basically always expects a CRS. For simple cropping, we have the `crop` method. But I agree that we should `reproject` should cope with such a use case.

**Comment by erikmannerfelt** [comment]
That makes sense, @atedstone! I do however believe many may use the reproject function like me. I (potentially erroneously) think of the `reproject()` function as "do stuff to fit the Raster to my needs".

Related question; does crop allow extending the bounds as well? Also, does crop/reproject allow for resampling to different grids?

If I have a raster with a 
```python
BoundingBox(left=1003, right=2013, top=1000, bottom=0)
```
 and I want to reproject/resample (not shift) this to 
```python
BoundingBox(left=1000, right=2010, top=1000, bottom=0)
```
 (perhaps to fit to another form of data / another raster), is this currently possible easily? How I would intuitively try it is:
```python

r.reproject(
    dst_bounds=BoundingBox(left=1000, right=2010, top=1000, bottom=0),
    resampling=rio.warp.Resampling.cubic_spline
)
```
but I'm not sure this would work at the moment.

**Comment by adehecq** [comment]
> That makes sense, @atedstone! I do however believe many may use the reproject function like me. I (potentially erroneously) think of the `reproject()` function as "do stuff to fit the Raster to my needs".
> 
It was not its initial goal, but now with the recent changes I made it kind of does all this, even replacing crop.

> Related question; does crop allow extending the bounds as well? Also, does crop/reproject allow for resampling to different grids?
`reproject` should be able to do that. `crop` will never do that as its primary goal is simply to remove pixels outside the specified extent.
> 
> If I have a raster with a
> 
> ```python
> BoundingBox(left=1003, right=2013, top=1000, bottom=0)
> ```
> 
> and I want to reproject/resample (not shift) this to
> 
> ```python
> BoundingBox(left=1000, right=2010, top=1000, bottom=0)
> ```
> 
> (perhaps to fit to another form of data / another raster), is this currently possible easily? How I would intuitively try it is:
> 
> ```python
> r.reproject(
>     dst_bounds=BoundingBox(left=1000, right=2010, top=1000, bottom=0),
>     resampling=rio.warp.Resampling.cubic_spline
> )
> ```
> 
> but I'm not sure this would work at the moment.
Yes, this should work ideally. I'll check why the CRS is needed.

Maybe it's worth indeed rediscussing the frames of crop and reproject, because it has been mixed a bit with recent changes.
I think `crop` should only keep the pixels that are within the provided extent, without doing any resampling. This is not quite the case because there is an option 'mode' that allows changing the pixel resolution.
`reproject` should be used for any kind of operation that requires resampling, like reprojecting to a new crs, but also cropping to an extent that is not a multiple of the resolution etc.
Do you agree with that?

**Comment by erikmannerfelt** [comment]
I think what you just said @adehecq makes perfect sense. Crop should always just trim edges of an otherwise unchanged array, and reproject can do all kinds of bending, inflation and expansion.

**Comment by rhugonnet** [comment]
I'd say we can keep "match_pixel"/"match_extent" functionality of `crop()` even if the "match_extent" is kind of redundant, while making `reproject()` more generic.

**Comment by erikmannerfelt** [comment]
As I mentioned in my accidental duplicate issue (#147 ), the line that messes this up is:
https://github.com/GlacioHack/GeoUtils/blob/2c38c0be21c9526ce8cbf5ef2495bdacc91feb4b/geoutils/georaster.py#L666

Because of that line, `dst_res` is required to allow `dst_bounds` to be used. This works as intended:
```python
r.reproject(dst_bounds=new_bounds, dst_res=r.res)
```

**Comment by adehecq** [comment]
Solved by #152. Happy @erikmannerfelt ? ;-)
